### PR TITLE
Adding a silencer to layouts

### LINKED ISSUE #285: Allow empty layouts
I think you mentioned this @philippjfr, but just to capture it. We should hide the bokeh warning since it can be very useful to create an empty layout and then add content to it.

screen shot 2019-03-06 at 12 52 42 pm

### PR DESCRIPTION
**Pull request by jsignell**
Not positive this is the right place - and will only work after https://github.com/bokeh/bokeh/pull/8739 lands

fixes #285

### CODE REVIEW

**Comment by jsignell** [comment]
This does affect bokeh after panel is imported - which is perhaps a bad and unintended consequence.

Screen Shot 2019-03-12 at 1 23 20 PM

**Comment by jsignell** [comment]
We could move the silence call into the init method to at least delay this side effect

**Comment by philippjfr** [comment]
Will have to wait until a new bokeh dev release is available.

File: panel/layout.py [hunk 1]
```diff
@@ -107,6 +111,7 @@ def _cleanup(self, root):
         super(Panel, self)._cleanup(root)
         for p in self.objects:
             p._cleanup(root)
+        silence(EMPTY_LAYOUT, False)
```
Comment (philippjfr) [inline panel/layout.py]: Could you explain this? This seems weird, I can imagine creating a panel, displaying it, deleting the cell or replacing the panel in the plot which triggers cleanup and then a subsequent display of the panel would again have the warning enabled again.

Comment (jsignell) [reply panel/layout.py]: I left that in by mistake. I was trying to make it cleanup after itself. But this isn't the right way.

**Comment by philippjfr** [comment]
It seems like some of my recent changes caused the warnings to be bypassed entirely, so before implementing this I think we need to reenable them since I assume at least some of them will be helpful for us.

**Comment by jsignell** [comment]
I am leaning towards my first approach of silencing on import rather than on init - what do you think @philippjfr

**Comment by philippjfr** [comment]
>I am leaning towards my first approach of silencing on import rather than on init - what do you think @philippjfr 

I agree, but currently it seems like the validation is not enabled at all with or without this PR. I think in refactoring some of the rendering code they are now bypassed or something.

**Comment by philippjfr** [comment]
Afaik warnings are already silenced as a side-effect of the way the rendering is done.

**Pull request closed without merging**